ocaml/parsing/ast_iterator.ml

```diff
@@ -797,15 +789,20 @@ let default_iterator =
 
     cases = (fun this l -> List.iter (this.case this) l);
     case =
-      (fun this {pc_lhs; pc_guard; pc_rhs} ->
+      (fun this {pc_lhs; pc_rhs} ->
          this.pat this pc_lhs;
-         iter_opt (this.guard this) pc_guard;
-         this.expr this pc_rhs
+         this.case_rhs this pc_rhs;
+      );
+    case_rhs =
+      (fun this -> function
+         | Psimple_rhs e -> this.expr this e
+         | Pboolean_guarded_rhs { pbg_guard; pbg_rhs } ->
+             this.expr this pbg_guard;
+             this.expr this pbg_rhs
+         | Ppattern_guarded_rhs { ppg_scrutinee; ppg_cases } ->
```

Comment:
I think this is missing iterating over the location. Can you spot-check the other iterators in this PR for the same?

Reply:
Good catch, fixed! The other similarly structured things are ast_mapper and untypeast: these do iterate (or map) over location.